README.md

```diff
@@ -35,7 +35,130 @@ even when:
 4. Clusters do not follow the rules of [Namespace
    Sameness](https://groups.google.com/forum/#!msg/kubernetes-sig-multicluster/jfDAMxFWlOg/9Z9O0mVpAgAJ)
 
-
+## Walk Through
+
+This walk through assumes:
+- A [management
+  cluster](https://cluster-api.sigs.k8s.io/reference/glossary.html#management-cluster)
+  exists, running the Cluster API.
+- Two [workload
+  clusters](https://cluster-api.sigs.k8s.io/user/quick-start.html#create-your-first-workload-cluster)
+  exist, with support for services type LoadBalancer. For the sake of this doc, assume
+  `cluster-a` and `cluster-b` exist and both of these Clusters belong to the
+  `dev-team` namespace on the management cluster.
+
+Note that if you're using `kind` that you'll need to BYO your own load balancer.
+Consider using [MetalLB](https://metallb.universe.tf).
+
+### Install Multi-cluster DNS on management cluster
+
+1. Install `GatewayDNS` CRD on the management cluster
+   ```
+   kubectl apply -f manifests/crds/connectivity.tanzu.vmware.com_gatewaydns.yaml
+   ```
+1. Install `capi-dns-controller` on the management cluster
+   ```
+   kubectl apply -f manifests/capi-dns-controller/deployment.yaml
+   ```
+
+### Install Multi-cluster DNS on each workload cluster
+
+1. Deploy `dns-server` controller
+   ```
+   kubectl apply -f manifests/dns-server/
+   ```
+1. Get IP address assigned to the DNS server service
+   ```
+   kubectl get service -n capi-dns dns-server -o=jsonpath='{.spec.clusterIP}'
+   ```
+1. Patch CoreDNS to forward `xcc.test` zone to the `dns-server`.
+   Note: the needed change is to add forwarding configuration for the
+   `xcc.test` zone. Preserve your Corefile's other configurations.
+   ```bash
+   kubectl -n kube-system patch configmap coredns \
+     --type=strategic --patch="$(cat <<EOF
+   data:
+     Corefile: |
+       .:53 {
+           errors
+           health {
+              lameduck 5s
+           }
+           ready
+           kubernetes cluster.local in-addr.arpa ip6.arpa {
+              pods insecure
+              fallthrough in-addr.arpa ip6.arpa
+              ttl 30
+           }
+           prometheus :9153
+           forward . /etc/resolv.conf
+           cache 30
+           loop
+           reload
+           loadbalance
+       }
+       xcc.test {
+           forward . <REPLACE_WITH_IP_OBTAINED_IN_PRIOR_STEP>
```

Comment:
Perhaps for copy-pasteability we could set the cluster ip to a variable and use that variable here?